Event / University Coordinate Fetching

Deadcoast opened the pull request:
Implemented functionality for the `getEvents` and `getUniversities` endpoints where the geolocation coordinates would be returned with the data.

Also fixed the following bugs:
- Error in `getUniversities` where the response was not being sent
- Error in `getUniversities` where the query was joining the states table on the wrong columns

The front-end types were also updated to include the new data

jamilgonzalez commented on src/api/getEvents.ts:
```diff
@@ -98,10 +95,14 @@ interface Event
-	eventLong: number, 		 	// find this 
-	schoolLat: number, 			// find this
-	schoolLong: number 			// find this
+	lattitude: number,			 	// find this
```
nit: latitude

jamilgonzalez commented on src/api/getEvents.ts:
```diff
@@ -118,17 +119,44 @@ interface EndpointReturn
+		{
+			coordinates.longitude = locations[0].longitude;
+		}
+	}
```
Probably safe to do something like

`locations.length > 0 ? locations : undefined`

or 

`const { latitude, longitude } = locations`

`latitude && longitude ? { lng: longitude, lat: latitude} : undefined`

janto393 replied:
Implemented safer practice:
```typescript
	// parse the data of the first match
	if ((locations !== undefined) && (locations.length > 0))
	{
		const { latitude, longitude } = locations[0];

		coordinates.latitude = latitude;
		coordinates.longitude = longitude;
	}
```

jamilgonzalez commented on src/api/getEvents.ts:
```diff
@@ -316,13 +347,20 @@ export async function getEvents(request: Request, response: Response, next: Call
 					};
+
+					// get event Coordinates
+					let eventCoords: Coordinates = await getLattitudeAndLongitude(event.address, event.city, event.state.name, event.zip);
```
nit: same as comment above about latitude spelling

jamilgonzalez commented:
Is there any progress on this PR? Comments have been sitting for a week.

Deadcoast commented:
> 
> 
> Is there any progress on this PR? Comments have been sitting for a week.

I didn't see the comments in this PR. I'll work on it tonight and get it fixed.

Deadcoast commented:
Fixed latitude spelling across all API endpoints. Packaged latitude and longitude into `Coordinate` type.

Fixed the fields passing into the `getUniversities` and `getEvents` endpoints from `schoolID` to `universityID`. Updated front-end types and API calls to accommodate change.

Implemented safer practice of location parsing:
```typescript
// parse the data of the first match
	if ((locations !== undefined) && (locations.length > 0))
	{
		const { latitude, longitude } = locations[0];

		coordinates.latitude = latitude;
		coordinates.longitude = longitude;
	}
```
